File: client/Models/PostModel.cs
namespace client
{
    public class PostModel
    {
        public PostModel()
        {

        }
        public PostModel(string key, string value)
        {
            Key = key;
            Value = value;
        }

        public string Key { get; set; }
        public string Value { get; set; }
    }
}

File: webservice/postModel.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

// PostModel handles the values sent from client
type PostModel struct {
	Key   string
	Value string
}

// ConvertToByteArray converts the model to a byte array
func (model *PostModel) ConvertToByteArray() []byte {
	var byteArray, err = json.Marshal(model)
	if err != nil {
		fmt.Println("Failed to convert model to byte array!")
		os.Exit(1)
	}

	return byteArray
}
